ballerina-shell/modules/shell-cli/src/main/java/io/ballerina/shell/cli/utils/IncompleteInputFinder.java

```diff
@@ -82,28 +129,98 @@ public Boolean transform(ListConstructorExpressionNode node) {
 
     @Override
     public Boolean transform(IfElseStatementNode node) {
-        return node.ifBody().isMissing() || node.ifBody().closeBraceToken().isMissing();
+        if (node.ifKeyword().isMissing()) {
+            return true;
+        }
+
+        return node.ifBody().openBraceToken().isMissing() || node.ifBody().closeBraceToken().isMissing();
     }
 
     @Override
     public Boolean transform(WhileStatementNode node) {
-        return node.whileBody().isMissing() || node.whileBody().closeBraceToken().isMissing();
+        if (node.whileKeyword().isMissing()) {
+            return true;
+        }
+
+        return node.whileBody().openBraceToken().isMissing() || node.whileBody().closeBraceToken().isMissing();
     }
 
     @Override
     public Boolean transform(ForEachStatementNode node) {
-        return node.blockStatement().isMissing() || node.blockStatement().closeBraceToken().isMissing();
+        if (node.forEachKeyword().isMissing() || node.inKeyword().isMissing()) {
+            return true;
+        }
+
+        return node.blockStatement().openBraceToken().isMissing()
+                || node.blockStatement().closeBraceToken().isMissing();
     }
 
     @Override
     public Boolean transform(VariableDeclarationNode node) {
-        if (node.equalsToken().isEmpty()) {
-            return false;
+        if (node.initializer().isPresent()) {
+            return node.equalsToken().get().isMissing() || node.initializer().get().apply(this)
+                    || node.typedBindingPattern().typeDescriptor().apply(this);
+        } else if (node.equalsToken().isPresent()) {
+            return node.equalsToken().get().isMissing() || node.typedBindingPattern().typeDescriptor().apply(this)
+                    || node.typedBindingPattern().bindingPattern().apply(this);
+        } else {
+            return node.typedBindingPattern().typeDescriptor().apply(this)
+                    || node.typedBindingPattern().bindingPattern().apply(this);
+        }
+    }
+
+    @Override
+    public Boolean transform(MappingBindingPatternNode node) {
+        return node.closeBrace().isMissing();
+    }
+
+    @Override
+    public Boolean transform(ModuleVariableDeclarationNode node) {
+        // Diagnostic check to avoid case : annotation not attached to construct
+        for (Diagnostic diagnostic : StreamSupport.stream(node.diagnostics().spliterator(), false)
+                .collect(Collectors.toList())) {
+            if (diagnostic.diagnosticInfo().code().equals("BCE0524")) {
+                return false;
+            }
         }
```

Comment:
Try to use the stream APIs

Reply:
Updated in 7d25e1dea188151fc8c5f84a1bf9156ce22ebc81